text/0436-gamelift-l2.md

```diff
@@ -0,0 +1,999 @@
+
+[Amazon GameLift](https://docs.aws.amazon.com/gamelift/latest/developerguide/gamelift-intro.html) is a service used to deploy, operate, and scale dedicated, low-cost servers in the cloud for session-based multiplayer games. Built on AWS global computing infrastructure, GameLift helps deliver high-performance, high-reliability game servers while dynamically scaling your resource usage to meet worldwide player demand.
+
+GameLift is composed of three main components:
+
+* GameLift FlexMatch which is a customizable matchmaking service for multiplayer games. With FlexMatch, you can build a custom set of rules that defines what a multiplayer match looks like for your game, and determines how to evaluate and select compatible players for each match. You can also customize key aspects of the matchmaking process to fit your game, including fine-tuning the matching algorithm.
+  
+* GameLift hosting for custom or realtime servers which helps you deploy, operate, and scale dedicated game servers. It regulates the resources needed to host games, finds available game servers to host new game sessions, and puts players into games.
+  
+* GameLift FleetIQ to optimize the use of low-cost Amazon Elastic Compute Cloud (Amazon EC2) Spot Instances for cloud-based game hosting. With GameLift FleetIQ, you can work directly with your hosting resources in Amazon EC2 and Amazon EC2 Auto Scaling while taking advantage of GameLift optimizations to deliver inexpensive, resilient game hosting for your players
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project. It allows you to define components for your matchmaking configuration or gmae server fleet management system.
+
+#### GameLift FlexMatch
+
+##### Defining a Matchmaking configuration
+
+FlexMatch is available both as a GameLift game hosting solution (including Realtime Servers) and as a standalone matchmaking service. To set up a FlexMatch matchmaker to process matchmaking requests, you have to create a matchmaking configuration based on a RuleSet.
+
+More details about matchmaking ruleSet are covered [below](#ruleSet).
+
+There is two types of Matchmaking configuration: through a queue system to let FlexMatch forms matches and uses the specified GameLift queue to start a game session for the match, and through a standalone version to let FlexMatch forms matches and returns match information in an event.
+
+
+Either a Standalone Matchmaking configuration
+
+```ts
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+
+new gamelift.MatchmakingConfiguration(this, 'Standalone Matchmaking', {
+  requestTimeouts: Duration.seconds(35)
+});
+```
+
+Either a Matchmaking configuration using a Queue
+```ts
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+
+const matchmaking = new gamelift.MatchmakingConfiguration(this, 'Queued Matchmaking', {
+  requestTimeouts: Duration.seconds(35)
+});
+
+const queue = new gamelift.Queue(this, 'GameLift Queue', {
```

Comment:
I think Queue deserves it's own section since there are some configurations there that would be good to document.

Reply:
What do you mean? Queue cannot be create in a standalone way I mean there is no sense of having a queue alone without a Fleet.

Reply:
I'm just meaning that there isn't much time spent explaining how to configure a queue / why you would want to. Looking at the CFN docs it looks like there is a lot to it.